sigmastate/src/main/scala/org/ergoplatform/ErgoBox.scala

```diff
@@ -77,6 +77,13 @@ class ErgoBox(
     */
   lazy val bytes: Array[Byte] = ErgoBox.sigmaSerializer.toBytes(this)
 
+  /** Creates a new box by updating some of the additional registers with the given new bindings.
+    * @param newBindings a map of the registers to be updated with new values
+    */
+  def withUpdatedRegisters(newBindings: AdditionalRegisters): ErgoBox = {
```

Comment:
Public API should literally enforce proper usage of it. This method is going in opposite direction, by returning incorrect box (with improper reference). As it is used only once in the testing code, maybe it is better to eliminate the method at all.

Reply:
Why "improper reference"?

Reply:
@aslesarenko transaction bytes (means reference field) would be changed after adding anything to a box created by the transaction.

Reply:
ok, removed.